### PR TITLE
Add FakePrague Egret R1 to fake backends

### PR DESCRIPTION
**Pull request by nonhermitian**
### Summary

Adds the Prague device to the fake backends

### Details and comments

Pulse defaults is not included as there is an error importing them.  However, this adds a new system with `cz` entangling gates, and thus is a nice addition to the fake backends.

### CODE REVIEW

File: releasenotes/notes/add_fake_prague-79f82b83c2e2329c.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    Adds the `FakePrague` Egret R1 backend to the FakeProvider.
```
Comment (mtreinish) [inline releasenotes/notes/add_fake_prague-79f82b83c2e2329c.yaml:4]:
```suggestion
    Adds the :class:`~.FakePrague` Egret R1 backend to the FakeProvider.
```

**Review by mtreinish — COMMENTED** [review]
LGTM, before I approve though can you add `FakePrague` to the list of backends in `FakeProviderForBackendV2` in `qiskit/providers/fake_provider/fake_provider.py`. That will use it in the test suite to validate everything works with it. Also we've been adding the fake backends off of `qiskit/provider/fake_provider/__init__.py` and including them in the autosummary for the docs.

**Comment by nonhermitian** [comment]
Ok, hopefully that addresses those issues.

**Review by mtreinish — APPROVED** [review]
LGTM, thanks for the quick updates.

**Comment by jakelishman** [comment]
@Mergifyio requeue

**Pull request merged**